src/main/kotlin/com/jetbrains/snakecharm/lang/psi/impl/refs/SmkRuleOrCheckpointNameReference.kt

```diff
@@ -51,7 +55,21 @@ class SmkRuleOrCheckpointNameReference(
         results.addAll(SmkCheckpointType(null, smkFile).resolveMember(name, element, ctx, myContext))
         results.addAll(collectModulesAndResolveThem(smkFile, element))
 
-        return results
+        val moduleRef = element.parentOfType<SmkUse>()?.getModuleName() as? SmkReferenceExpression
+        val itIsModuleMameReference = element.parent is SmkUse//.elementType == SmkElementTypes.USE_IMPORTED_RULES_NAMES
+        val parentIsImportedRuleNames = element.parent.elementType == SmkElementTypes.USE_IMPORTED_RULES_NAMES
```

Comment:
Could  you refactor this to replace AST node type check with using instance of corresponding PSI interface?  + cleanup the previous line

Reply:
Done